Sources/VersionCheck/VersionCheck.swift

```diff
@@ -0,0 +1,99 @@
+    /// the backend has succeeded, no additional fetch requests to the backend will be allowed until the
+    /// minimum fetch interval expires. Note that you can override this default on a per-fetch request
+    /// basis using `RemoteConfig.fetch(withExpirationDuration:)`. For example, setting
+    /// the expiration duration to 0 in the fetch request will override the `minimumFetchInterval` and
+    /// allow the request to proceed.
+    /// - Returns: ``VersionCheck/UpdateResult``
+    /// - Throws: ``VersionCheck/CustomError``
+    public static func validateVersion(currentVersionKey: String,
+                                minSupportedVersionKey: String,
+                                currentAppVersion: String,
+                                minimumFetchInterval: Double? = nil,
+                                defaultValues: [String: Any?]? = nil) async throws -> UpdateResult {
+        let remoteConfig = RemoteConfig.remoteConfig()
+        VersionCheck.customizeSettings(remoteConfig: remoteConfig,
+                                       minimumFetchInterval: minimumFetchInterval,
+                                       defaultValues: defaultValues)
+        let fetchedRemoteConfig = await VersionCheck.fetchRemoteConfig(remoteConfig: remoteConfig)
+        let activatedRemoteConfig = try await VersionCheck.activeRemoteConfig(remoteConfig: fetchedRemoteConfig.get())
+        let currentValidVersion = try VersionCheck.getStringValueForeKey(remoteConfig: activatedRemoteConfig.get(), key: currentVersionKey)
+        let minSupportedVersion = try VersionCheck.getStringValueForeKey(remoteConfig: activatedRemoteConfig.get(), key: minSupportedVersionKey)
+        return  VersionCheck.isThereAnyUpdate(currentValidVersionValue: currentValidVersion,
+                                              minSupportedVersionValue: minSupportedVersion,
+                                              currentAppVersion: currentAppVersion)
+    }
+}
+
+extension VersionCheck {
+    private static func customizeSettings(remoteConfig: RemoteConfig,
+                                          minimumFetchInterval: Double?,
+                                          defaultValues: [String: Any?]?) {
+        if let minimumFetchInterval = minimumFetchInterval {
+            let settings = RemoteConfigSettings()
+            settings.minimumFetchInterval = minimumFetchInterval
+            remoteConfig.configSettings = settings
+        }
+        if let defaultValues = defaultValues as? [String: NSObject] {
+            remoteConfig.setDefaults(defaultValues)
+        }
+    }
+    static func getStringValueForeKey(remoteConfig: RemoteConfig, key: String) throws -> String {
+        guard let value = remoteConfig.configValue(forKey: "\(key)").stringValue
+        else { throw CustomError.noValueFound(key: key)}
+        return value
+    }
+    static func isThereAnyUpdate(currentValidVersionValue: String,
```

Comment:
As we talked about, maybe change currentValidVersionValue to latestVersion or something like that

Reply:
Maybe something with "fetched"  so we know it comes from firebase

Reply:
done